### PR TITLE
[Spring] resolves #16054, resolves #15088 add vendor extension x-spring-provide-args

### LINKED ISSUE #15088: How to add parameters generated by HandlerMethodArgumentResolver?
Can you please tell me if it is possible to add parameters to the generated controllers so that Spring creates them at runtime using implementations of the HandlerMethodArgumentResolver class?

I couldn't find anything relevant on this subject in Google, except for Pageable examples.

### LINKED ISSUE #16054: [REQ] Add a new config for spring controller generation to add @Headers into the API definition
### Is your feature request related to a problem? Please describe.

When using openapi generator with properties:
    generatorName = "spring"
    library = "spring-boot"

The API definition is generated using the following strategy:

`    CompletableFuture<ResponseEntity<DtoResponse>> methodName(
        @Parameter(name = "DtoRequest", description = "", required = true) @Valid @RequestBody DtoRequest req);
`

It would be great to add a possibility to add into the API definition the parameter headers (spring @RequestHeaders):

`
    CompletableFuture<ResponseEntity<DtoResponse>> methodName(
        @Parameter(name = "DtoRequest", description = "", required = true) @Valid @RequestBody DtoRequest req, 
        @RequestHeader HttpHeaders requestHeaders);
`

## Describe the solution you'd like

Add a new property that should be consumed only if 
    library = "spring-boot"
Ex:
    configOptions = [
            useSpringBoot3                : "true",
            addRequestHeadersToAPI: "true", #Default will be false, same behaviour as today
            (....)
    ]

That will add @RequestHeader HttpHeaders requestHeaders into the API definition parameters. 

## Describe alternatives you've considered

I've updated myself the mustache templates to achieve this - it's working. However this will add technical debt for the future. Every time I want to upgrade openapi generator version I will have to pull the new mustache templates and apply the same tranformation

## Additional context

Be able to easily use request HttpHeaders in a controller that is generated or templated by open api generator

### PR DESCRIPTION
**Pull request by dabdirb**
to fix #16054
to fix #15088 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by dabdirb** [comment]
this PR is to fix issue 16054, by add vendor extenstion x-spring-provide-args

**Comment by wing328** [comment]
cc @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09) @martin-mfg (2023/08)

**Comment by martin-mfg** [comment]
Hi @dabdirb, thank you for the PR!

I have two comments:
- What's the purpose of this line in `petstore-with-spring-provide-args.yaml`?
   ```yaml
   #        - '@org.springframework.security.oauth2.client.annotation.RegisteredOAuth2AuthorizedClient("okta") org.springframework.security.oauth2.client.OAuth2AuthorizedClient authorizedClient'
   ```
   I tried uncommenting it, but then the generated code doesn't compile anymore.
- I think your new sample should be added to some pipeline, probably to `.github/workflows/samples-spring.yaml`.

**Comment by dabdirb** [comment]
Hello, @martin-mfg 

* Whats the purpose, i want add use case of this vendor extension.
 ```
@org.springframework.security.oauth2.client.annotation.RegisteredOAuth2AuthorizedClient("okta") org.springframework.security.oauth2.client.OAuth2AuthorizedClient authorizedClient
```
This line is a use case when our service need access another server/service by webclient, with oauth token, we need a bean that autowired by spring framework, the oauth configuration, such as cliend-id and client-secret, is read from property file.
```
spring.security.oauth2.client.registration.okta.client-id=xxxxxxxxxx
spring.security.oauth2.client.registration.okta.client-secret=yyyyyyyyyyyyyyyyyyyy
```

please refer https://docs.spring.io/spring-security/reference/reactive/oauth2/client/authorized-clients.html

It can not be compile success because we should add spring security dependency in the POM.
I removed it from sample yaml to avoid confusing.

* Added  "samples/server/petstore/springboot-spring-provide-args"  into  .github/workflows/samples-spring.yaml

* "implicitHeadersRegex: ^Version.*" was removed

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/VendorExtension.java [hunk 1]
```diff
@@ -7,6 +7,7 @@ public enum VendorExtension {
 
     X_IMPLEMENTS("x-implements", ExtensionLevel.MODEL, "Ability to specify interfaces that model must implements", "empty array"),
     X_SPRING_PAGINATED("x-spring-paginated", ExtensionLevel.OPERATION, "Add org.springframework.data.domain.Pageable to controller method. Can be used to handle page & size query parameters", "false"),
+    X_SPRING_PROVIDE_ARGS("x-spring-provide-args", ExtensionLevel.OPERATION, "Added spring provide arguments in controller", "empty array"),
```
[the remark this answers is not in the document]
Comment (dabdirb) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/VendorExtension.java]: @welshm , this description is better, i updated the VendorExtension.java

**Review by welshm — APPROVED** [review]

**Pull request merged**